app/src/main/kotlin/com/wire/android/ui/settings/devices/SelfDevicesViewModel.kt

```diff
@@ -53,37 +54,47 @@ class SelfDevicesViewModel @Inject constructor(
         private set
 
     init {
-        // this will cause the list to be refreshed
-        loadClientsList()
-        observeClientList()
+        fetchAndObserveClientList()
     }
 
-    private fun observeClientList() {
+    private fun fetchAndObserveClientList() {
         viewModelScope.launch {
-            observeClientList(currentAccountId).collect { result ->
-                state = when (result) {
-                    is ObserveClientsByUserIdUseCase.Result.Failure -> state.copy(isLoadingClientsList = false)
-                    is ObserveClientsByUserIdUseCase.Result.Success -> {
-                        val currentClientId = currentClientIdUseCase().firstOrNull()
-                        val e2eiCertificates = getUserE2eiCertificates(currentAccountId)
-                        state.copy(
-                            isLoadingClientsList = false,
-                            currentDevice = result.clients
-                                .firstOrNull { it.id == currentClientId }
-                                ?.let { Device(it, e2eiCertificates[it.id.value]) },
-                            deviceList = result.clients
-                                .filter { it.id != currentClientId }
-                                .map { Device(it, e2eiCertificates[it.id.value]) }
-                        )
-                    }
-                }
+            fetchSelfClientsFromRemote() // this will cause the list to be refreshed
+            observeClientList(currentAccountId)
+                .collect(::intoState)
+        }
+    }
+
+    private suspend fun intoState(result: ObserveClientsByUserIdUseCase.Result) {
+        state = when (result) {
+            is ObserveClientsByUserIdUseCase.Result.Success -> {
+                val currentClientId = currentClientIdUseCase().firstOrNull()
+                val e2eiCertificates = getUserE2eiCertificates(currentAccountId)
+                state.copy(
+                    isLoadingClientsList = false,
+                    currentDevice = result.clients
+                        .firstOrNull { it.id == currentClientId }
+                        ?.let { Device(it, e2eiCertificates[it.id.value]) },
+                    deviceList = result.clients
+                        .filter { it.id != currentClientId }
+                        .map { Device(it, e2eiCertificates[it.id.value]) }
+                )
+            }
+            is ObserveClientsByUserIdUseCase.Result.Failure -> {
+                state.copy(isLoadingClientsList = false, error = SelfDevicesState.Error.InitError)
```

Comment:
If we have the clients list locally, there is no need to display the error message

Reply:
I just thought it would be a nice additional safety, but maybe you're right. I removed these errors from self devices screen, just double checked and we do get full client data for self user with `Event.User.NewClient` and during slow sync (also initial slow sync) when updating self supported protocols. For very old devices, they cannot be legal hold ones as it's not yet used so we can quite safely assume that they are either permanent or temporary, both can be removed, so it's fine.